src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -675,36 +675,28 @@ private void AddMigratorBar()
             _root.Children.Insert(0, _migratorBar);
         }
 
-#pragma warning disable IDE1006 // Naming Styles
-        private void packageRestoreManager_PackagesMissingStatusChanged(object sender, PackagesMissingStatusEventArgs e)
-#pragma warning restore IDE1006 // Naming Styles
+        private void PackageRestoreManager_PackagesMissingStatusChanged(object sender, PackagesMissingStatusEventArgs e)
         {
-            // make sure update happens on the UI thread.
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            // TODO: PackageRestoreManager fires this event even when solution is closed.
+            // Don't do anything if solution is closed.
+            // Add MissingPackageStatus to keep previous packageMissing status to avoid unnecessarily refresh
+            // only when package is missing last time and is not missing this time, we need to refresh
+            if (!e.PackagesMissing && _missingPackageStatus)
             {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                // TODO: PackageRestoreManager fires this event even when solution is closed.
-                // Don't do anything if solution is closed.
-                // Add MissingPackageStatus to keep previous packageMissing status to avoid unnecessarily refresh
-                // only when package is missing last time and is not missing this time, we need to refresh
-                if (!e.PackagesMissing && _missingPackageStatus)
-                {
-                    await UpdateAfterPackagesMissingStatusChangedAsync();
-                }
+                UpdateAfterPackagesMissingStatusChanged();
+            }
 
-                _missingPackageStatus = e.PackagesMissing;
-            });
+            _missingPackageStatus = e.PackagesMissing;
         }
 
         // Refresh the UI after packages are restored.
         // Note that the PackagesMissingStatusChanged event can be fired from a non-UI thread in one case:
         // the VsSolutionManager.Init() method, which is scheduled on the thread pool.
-        private async ValueTask UpdateAfterPackagesMissingStatusChangedAsync()
+        private void UpdateAfterPackagesMissingStatusChanged()
         {
-            VSThreadHelper.ThrowIfNotOnUIThread();
             var timeSinceLastRefresh = GetTimeSinceLastRefreshAndRestart();
-            await RefreshAsync();
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () => await RefreshAsync())
+                .PostOnFailure(nameof(PackageManagerControl), nameof(UpdateAfterPackagesMissingStatusChanged));
             EmitRefreshEvent(timeSinceLastRefresh, RefreshOperationSource.PackagesMissingStatusChanged, RefreshOperationStatus.Success);
             _packageDetail.Refresh();
```

Comment:
This PR changes the behaviour of this method. Previously `EmitRefreshEvent` and `_packageDetail.Refresh()` are only called after `RefreshAsync` returns. This PR changes it so they are called before `RefreshAsync` has probably even been started. The `EmitRefreshEvent` looks particularly weird, and it's saying the refresh was successful, before it even started?

Perhaps what Scott means is that `_packageDetail.Refresh()` needs to be called inside the anonymous delegate passed to `jtf.RunAsync(...)`?

Reply:
Yes that is what I was trying to say, thank you.

Reply:
> Previously EmitRefreshEvent and _packageDetail.Refresh() are only called after RefreshAsync returns

That's actually inconsistent compared what the rest of the refreshes are doing. 
Most of the other refreshes are emitting the event before the refresh gets kicked off. I have changed this method to do that as well. 

> Perhaps what Scott means is that _packageDetail.Refresh() needs to be called inside the anonymous delegate passed to jtf.RunAsync(...)?

Wouldn't that would make it worse?
_packageDetail.Refresh would only be triggered after the refresh of the scroll list is complete, but this way we'd kick off the work asynchronously.

Reply:
So looking at RefreshAsync(), it calls _packageDetail.Refresh as well, do we really need it to be called twice?

Reply:
Good call @sbanni 

Fixed.